src/schema/basic_slots.yaml

```diff
@@ -31,7 +31,11 @@ slots:
     description: >-
       A unique identifier for a thing.
       Must be either a CURIE shorthand for a URI or a complete URI
-    #required: false # for now we are setting this to false until we develop an id template
+    notes:
+      - "relocated YAML comment: for now we are setting this to false until we develop an id template"
+      - "Doesn't 'identifier: true' imply 'required: true'?"
+      - "abstracted pattern: prefix:typecode-authshoulder-blade(.version)?(_seqsuffix)?"
+    pattern: ^nmdc:[A-Za-z0-9]{1,6}-[A-Za-z0-9]{1,6}-[A-Za-z0-9]{1,6}(\.[A-Za-z0-9]{1,6})?(_.+)?$
```

Comment:
blade should be open-ended length, i.e. `[A-Za-z0-9]{1,}`

Reply:
Hm, I don't think I'm understanding the purpose of the last grouping `_.+`?

Reply:
I think the regex pattern for evaluating base32 encodings is `[A-Z2-7]+=*`

Base32 uses A-Z and 2-7 for the encoding, and adds a padding character = to get a multiple of 8 characters. And according to @dwinston's [gist](https://gist.github.com/dwinston/083a1cb508bbff21d055e7613f3ac02f) the blade is a base32 encoding of any 64 bit integer. So we can change the blade portion of the above regex do you think?

Reply:
I think we might want to standardize our naming of minting authorities, and further constrict our regex?

Reply:
> Hm, I don't think I'm understanding the purpose of the last grouping _.+?

That's for contig identifiers, which may be part of NMDC protein identification identifiers

From Paul Piehowski: nmdc:mga0wn63_1070_c1_4056_7928

(_.+)? would correspond to _1070_c1_4056_7928 in this case

Reply:
Maybe we should constrain that `.` any character wildcard

Reply:
> I think the regex pattern for evaluating base32 encodings is [A-Z2-7]+=*

OK, this requires more discussion, esp. re. receiving and emitting characters. I would like keep the pattern as simple as possible. If we need to accept lower case from an identifier minter, then we should have a canonical convert that is applied before inserting anything into a NMDC backend.

thttps://www.crockford.com/base32.html

Reply:
Proposed changes to regex:

* type code
  *  all lowercase characters, no digits
  * soft acceptance of minimum length of 3 characters, but no widely accepted
  * regex: `[a-z]{1,6}`
* shoulder
  * no need to stick to `{number}{characters}{number}` pattern for shoulder since we are using hyphens for separating components of the identifiers out
  * all lowercase between padding numbers
* blade
  * open ended length `[A-Za-z0-9]{1,}`

Reply:
Just to give some concrete examples... 

A contig ID might look like...

```
nmdc:mgasm-00-wn63_1070_c1
```

and a gene/protein ID might look like...

```
nmdc:mgann-00-mn1c6_1070_c1_4056_7928
```

I think the hyphen's are fine but we should confirm this doesn't cause any tool problems.

Note that in this example the shoulder is different between these even though the gene ID is coming from that contig in this example.

I'll give some examples of a MAG bin as well.

Reply:
Hmm.  The MAGs/bins are kind of tricky.   We have bin IDs.  Right now we don't really do much with these in the system.  So they are just bin.1, bin.2, etc.  So we should probably at a minimum, rename these to include the activity ID...  something like....

nmdc:mgmag-00-x012.1_bin_1

In the various output files, the workflow currently uses the original contig IDs.

So they currently look like...

```
>nmdc:mga0mn1c69_7_c1
```

So I'm not sure what to do with this.  I suppose it could be renamed as well.  So..

```
> nmdc:mgmag-00-x012.1_7_c1
```

This is a little weird because we would have to different types of suffixes on this ID.